src/program/lwaftr/migrate_configuration/migrate_configuration.lua

```diff
@@ -270,13 +270,18 @@ function load_binding_table(file)
    return parse_binding_table(Parser.new(source:as_text_stream()))
 end
 
+
 local function config_to_string(schema, conf)
    if type(schema) == "string" then
       schema = yang.load_schema_by_name(schema)
    end
-   local memfile = util.string_output_file()
-   yang.print_config_for_schema(schema, conf, memfile)
-   return memfile:flush()
+   -- To keep memory usage as low as possible write it out to a temp file.
+   local temp = io.open("/tmp/migrate-configuration-temp", "w")
+   yang.print_config_for_schema(schema, conf, temp)
+   temp:close()
+   conf = io.open("/tmp/migrate-configuration-temp", "r"):read("*a")
+   os.remove("/tmp/migrate-configuration-temp")
+   return conf
```

Comment:
Humm, I think this one might not work -- if there are multiple migrations happening at the same time, this will lead to sadness.  Does the memory issue occur in normal operation, or just when migrating legacy configs?  If it's just when migrating legacy configs, perhaps the `collectgarbage` call below is sufficient.